packages/models-library/tests/test_service_settings_labels.py

```diff
@@ -147,3 +155,173 @@ def test_raises_error_wrong_restart_policy() -> None:
+            {container_name: container_whitelist}
+        ),
+        "simcore.service.containers-allowed-outgoing-internet": json.dumps(
+            [container_name]
+        ),
+        "simcore.service.compose-spec": json.dumps(compose_spec),
+        "simcore.service.container-http-entrypoint": container_name,
+    }
+
+    instance = DynamicSidecarServiceLabels.parse_raw(json.dumps(dict_data))
+    assert (
+        instance.containers_allowed_outgoing_whitelist[container_name][0]
+        == expected_host_whitelist_policy
+    )
+
+
+def test_container_outgoing_whitelist_and_container_allow_internet_without_compose_spec():
+    dict_data = {
+        "simcore.service.containers-allowed-outgoing-whitelist": json.dumps(
+            {
+                DEFAULT_SINGLE_SERVICE_NAME: [
+                    {
+                        "hostname": "a-host",
+                        "tcp_ports": [12132, {"lower": 12, "upper": 2334}],
+                    }
+                ]
+            }
+        ),
+        "simcore.service.containers-allowed-outgoing-internet": json.dumps(
+            [DEFAULT_SINGLE_SERVICE_NAME]
+        ),
+    }
+    assert DynamicSidecarServiceLabels.parse_raw(json.dumps(dict_data))
+
+
+def test_container_allow_internet_no_compose_spec_not_ok():
+    dict_data = {
+        "simcore.service.containers-allowed-outgoing-internet": json.dumps(["hoho"]),
+    }
+    with pytest.raises(ValidationError) as exec_info:
+        assert DynamicSidecarServiceLabels.parse_raw(json.dumps(dict_data))
+
+    assert "Expected only 1 entry 'container' not '{'hoho'}" in f"{exec_info.value}"
+
+
+def test_container_allow_internet_compose_spec_not_ok():
+    container_name = "test_container"
+    compose_spec: dict[str, Any] = {"services": {container_name: None}}
+    dict_data = {
+        "simcore.service.compose-spec": json.dumps(compose_spec),
+        "simcore.service.containers-allowed-outgoing-internet": json.dumps(["hoho"]),
+    }
+    with pytest.raises(ValidationError) as exec_info:
+        assert DynamicSidecarServiceLabels.parse_raw(json.dumps(dict_data))
+
+    assert f"container='hoho' not found in {compose_spec=}" in f"{exec_info.value}"
+
+
+def test_container_outgoing_whitelist_no_compose_spec_not_ok():
+    dict_data = {
+        "simcore.service.containers-allowed-outgoing-whitelist": json.dumps(
+            {
+                "container_name": [
+                    {
+                        "hostname": "a-host",
+                        "tcp_ports": [12132, {"lower": 12, "upper": 2334}],
+                    }
+                ]
+            }
+        ),
+    }
+    with pytest.raises(ValidationError) as exec_info:
+        assert DynamicSidecarServiceLabels.parse_raw(json.dumps(dict_data))
+    assert (
+        f"Expected only one entry '{DEFAULT_SINGLE_SERVICE_NAME}' not 'container_name'"
+        in f"{exec_info.value}"
+    )
+
+
+def test_container_outgoing_whitelist_compose_spec_not_ok():
```

Comment:
Q: all ``test_*_not_ok`` rely in the fact that compose-specs are missing. What about the cases i mentioned earlier of ``-whitelist`` and ``-internet`` listing missing?  Is there a combination not allowed as I refer above?